src/auto/cell_layout.rs

```diff
@@ -94,9 +97,31 @@ impl<O: IsA<CellLayout>> CellLayoutExt for O {
     //    unsafe { TODO: call ffi::gtk_cell_layout_set_attributes() }
     //}
 
-    //fn set_cell_data_func<'a, P: IsA<CellRenderer>, Q: Into<Option<&'a /*Unimplemented*/CellLayoutDataFunc>>, R: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, cell: &P, func: Q, func_data: R, destroy: /*Unknown conversion*//*Unimplemented*/DestroyNotify) {
-    //    unsafe { TODO: call ffi::gtk_cell_layout_set_cell_data_func() }
-    //}
+    fn set_cell_data_func<P: IsA<CellRenderer>, Q: Fn(&CellLayout, &CellRenderer, &TreeModel, &TreeIter) + Send + Sync + 'static, R: Into<Option<Q>>>(&self, cell: &P, func: R) {
+        let func = func.into();
+        let func_data: Box_<Option<Q>> = Box::new(func.into());
+        unsafe extern "C" fn func_func<P: IsA<CellRenderer>, Q: Fn(&CellLayout, &CellRenderer, &TreeModel, &TreeIter) + Send + Sync + 'static>(cell_layout: *mut ffi::GtkCellLayout, cell: *mut ffi::GtkCellRenderer, tree_model: *mut ffi::GtkTreeModel, iter: *mut ffi::GtkTreeIter, data: glib_ffi::gpointer) {
+            let cell_layout = from_glib_borrow(cell_layout);
+            let cell = from_glib_borrow(cell);
+            let tree_model = from_glib_borrow(tree_model);
+            let iter = from_glib_borrow(iter);
+            let callback: &Box_<Option<Q>> = &*(data as *mut _);
+            if let Some(ref callback) = **callback {
+                callback(&cell_layout, &cell, &tree_model, &iter)
+            } else {
+                panic!("cannot get closure...")
+            };
+        }
+        let func = if func_data.is_some() { Some(func_func::<P, Q> as _) } else { None };
```

Comment:
Why does this have a `as _` btw? That case should not have any effect

Reply:
If I don't put it:

```
error[E0308]: mismatched types
   --> src/auto/cell_layout.rs:122:117
    |
122 |             ffi::gtk_cell_layout_set_cell_data_func(self.as_ref().to_glib_none().0, cell.as_ref().to_glib_none().0, func, Box::into_raw(super_callback0) as *mut _, destroy_call4);
    |                                                                                                                     ^^^^ expected fn pointer, found fn item
    |
    = note: expected type `std::option::Option<unsafe extern "C" fn(*mut ffi::GtkCellLayout, *mut ffi::GtkCellRenderer, *mut ffi::GtkTreeModel, *mut ffi::GtkTreeIter, *mut libc::c_void)>`
               found type `std::option::Option<unsafe extern "C" fn(*mut ffi::GtkCellLayout, *mut ffi::GtkCellRenderer, *mut ffi::GtkTreeModel, *mut ffi::GtkTreeIter, *mut libc::c_void) {<O as auto::cell_layout::CellLayoutExt>::set_cell_data_func::func_func::<P, Q>}>`

error: aborting due to previous error
```

Reply:
Wat? This probably works if you directly put `Some(the_actual_function::<P, Q>)` here instead of first going via a variable, can you confirm?